CodeReview/src/commands/CFocus.cpp

```diff
@@ -0,0 +1,184 @@
+
+	if (!focusInformationFound){
+		currentStep_ = 0;
+		focusInformationFound = focusInformationForStep(currentStep_, focusInformation);
+	}
+
+	if (!focusInformationFound)
+		return new Interaction::CommandResult{};
+
+	auto focusItem = CodeReviewManager::instance().overlayForNodeReviews(focusInformation.node_);
+
+	switch (focusInformation.type_) {
+		case FocusInformation::Center:
+			Visualization::VisualizationManager::instance().mainView()->
+					centerOn(focusItem);
+			break;
+		case FocusInformation::Highlight:
+		{
+			auto overlay = new Visualization::HighlightOverlay{focusItem,
+					Visualization::HighlightOverlay::itemStyles().get("focus")};
+			focusItem->addOverlay(overlay, "focusOverlay");
+			break;
+		}
+		default:
+			Q_ASSERT(false);
+			break;
+	}
+
+	currentStep_++;
+
+	return new Interaction::CommandResult{};
+}
+
+QList<Interaction::CommandSuggestion*> CFocus::suggest(Visualization::Item*, Visualization::Item*,
+		const QString& textSoFar, const std::unique_ptr<Visualization::Cursor>&)
+{
+	if (name().startsWith(textSoFar))
+		return {new Interaction::CommandSuggestion{name()}};
+	return {};
+}
+
+CFocus::FocusInformation CFocus::extractFocusInformation(QString line)
+{
+	if (line.startsWith("[focus"))
+	{
+		auto endOfCommand = line.indexOf("]");
+		// subtract one because of starting [
+		auto commandSize = endOfCommand - 1;
+		auto commandText = line.mid(1, commandSize);
+		QStringList commandTokens = commandText.split("|");
+
+		// remove first token which is "focus"
+		commandTokens.pop_front();
+
+		FocusInformation focusInformation;
+
+		if (!commandTokens.isEmpty())
+		{
+			auto token = commandTokens.takeFirst();
+			if (token.startsWith("hi"))
+				focusInformation.type_ = FocusInformation::FocusType::Highlight;
+			else if (token.startsWith("ce"))
+				focusInformation.type_ = FocusInformation::FocusType::Center;
+		}
+
+		if (!commandTokens.isEmpty())
+		{
+			auto stepToken = commandTokens.takeFirst();
+			focusInformation.step_ = stepToken.toInt();
+		}
+		return focusInformation;
+	}
+
+	return {};
+}
+
+void CFocus::extractFocusInformation()
+{
+	for (NodeReviews* nodeReviews : *CodeReviewManager::instance().nodeReviewsList())
+	{
+		if (nodeReviews->reviewComments()->isEmpty())
+			continue;
+
+		auto firstComment = nodeReviews->reviewComments()->first()->commentNode();
+
+		if (firstComment->lines()->isEmpty())
+			continue;
+
+		auto firstLine = firstComment->lines()->first()->get();
+
+		auto focusInformation = extractFocusInformation(firstLine);
+
+		if (focusInformation.type_ != FocusInformation::None)
+		{
+			focusInformation.node_ = nodeReviews;
+			addFocusInformation(focusInformation);
+		}
+	}
+}
+
+void CFocus::addFocusInformation(FocusInformation focusInformation)
+{
+	focusList_.insert(focusInformation.step_, focusInformation);
+}
+
+bool CFocus::focusInformationForStep(int step, FocusInformation& focusInformation)
```

Comment:
You don't need this function at all. simply use `QHash::value(step)` which will return a default constructed focus (so invalid) if the element is not found.